wom/src/main/scala/wom/types/WomPrimitiveType.scala

```diff
@@ -1,11 +1,13 @@
 package wom.types
 
-abstract class WomPrimitiveType extends WomType {
+trait WomPrimitiveType extends WomType {
   lazy val coercionMap: Map[WomType, Seq[WomType]] = Map(
     // From type -> To type
     WomAnyType -> Seq(WomStringType, WomIntegerType, WomFloatType, WomSingleFileType, WomBooleanType),
     WomStringType -> Seq(WomStringType, WomIntegerType, WomFloatType, WomSingleFileType, WomBooleanType),
+    WomSingleDirectoryType -> Seq(WomStringType, WomSingleDirectoryType),
     WomSingleFileType -> Seq(WomStringType, WomSingleFileType),
+    WomGlobFileType -> Seq(WomStringType, WomGlobFileType),
```

Comment:
Heads up: `WomGlobFile`s are only ever returned by `evaluateFiles` as a placeholder and never actually passed around in Cromwell. 
- When PAPI2 removes the need for `evaluateFiles`, we should be able to remove `WomGlobFile` too
- If we're trying to coerce to or from them, or store them in any value field in memory or in the database, we're probably doing something wrong.